### PR TITLE
merge work sprint from devel into main

### PR DESCRIPTION
**Pull request by mixmasala**
The devel branch fixes issues with the consensus protocol, removes duplicated abstractions in core/pki, improves container based workflow to use podman so that it can be used rootlessly, fixes issues with client and mix authentication, and removes the options for case insensitive session IDs.

### CODE REVIEW

**Review by david415 — CHANGES_REQUESTED** [review]
please also update katzenpost/docs/specs/pki.rst... if you don't want to do this in this same pull request then please open a ticket to update the specification documents in order to resolve this PR request for changes.

**Comment by david415** [comment]
looks good to me... but we do need to keep the specification documents up to date... so update the pki design doc and include the details about your wire protocol commands and the phases of the dirauth voting protocol.

File: core/sphinx/path/path.go [hunk 1]
```diff
@@ -107,40 +107,49 @@ func selectHops(rng *mRand.Rand, doc *pki.Document, src, dst *pki.MixDescriptor,
...
+	idHash = dst.IdentityKey.Sum256()
+	dstLayer, err := doc.GetMixLayer(&idHash)
+	if err != nil {
+		return nil, err
+	}
 	if isForward {
-		if dst.Layer != pki.LayerProvider {
-			return nil, fmt.Errorf("path: invalid destination layer: %v", dst.Layer)
+		if !dst.Provider {
+			return nil, fmt.Errorf("path: invalid destination (non provider): %x", dst.IdentityKey.Sum256())
```
[the remark this answers is not in the document]
Comment (mixmasala) [reply core/sphinx/path/path.go:122]: Endpoint?

File: authority/cmd/voting/authority.toml.sample [hunk 2]
```diff
@@ -4,8 +4,7 @@
 # The Authority section contains mandatory information.
 #
 
-[Authority]
-
+[Server]
```
[the remark this answers is not in the document]
Comment (mixmasala) [reply authority/cmd/voting/authority.toml.sample:7]: Because the Authority type is defined elsewhere for the Voting PKI peers

**Comment by mixmasala** [comment]
https://github.com/katzenpost/katzenpost/issues/149

File: authority/cmd/voting/authority.toml.sample (hunk 2 above)
[the remark this answers is not in the document]
Comment (mixmasala) [reply authority/cmd/voting/authority.toml.sample:7]: Because the Authority type is defined elsewhere for the Voting PKI peers.

**Review by david415 — APPROVED** [review]

File: core/sphinx/path/path.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (yippyskippy) [reply core/sphinx/path/path.go:122]: IIUC include `mix` so `endpoint mix` maybe?

**Comment by mixmasala** [comment]
Added fixes for #150 #151 which were caught because we now generate the testing mixnet configuration at build time from parameters that were more aggressive (shorter per-hop delays)

File: core/sphinx/path/path.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (leif) [reply core/sphinx/path/path.go:122]: we should change the term provider, but it's in many places (including right here already) and we should think more about the right word.

**Pull request merged**